figures/pipeline/enrollment_metrics.py

```diff
@@ -78,36 +78,28 @@ def bulk_calculate_course_progress_data(course_id, date_for=None):
 
     """
     progress_percentages = []
-
     if not date_for:
         date_for = datetime.utcnow().replace(tzinfo=utc).date()
 
     site = get_site_for_course(course_id)
     if not site:
         raise UnlinkedCourseError('No site found for course "{}"'.format(course_id))
 
-    course_sm = get_student_modules_for_course_in_site(site=site,
-                                                       course_id=course_id)
+    # We might be able to make this more efficient by finding only the learners
+    # in the course with StudentModule (SM) records then get their course
+    # enrollment (CE) records as we can ignore any learners without SM records
+    # since that means they don't have any course progress
     for ce in course_enrollments_for_course(course_id):
-        metrics = collect_metrics_for_enrollment(site=site,
-                                                 course_enrollment=ce,
-                                                 course_sm=course_sm,
-                                                 date_for=date_for)
-        if metrics:
-            progress_percentages.append(metrics.progress_percent)
-        else:
-            # Log this for troubleshooting
-            error_data = dict(
-                msg=('Unable to create or retrieve enrollment metrics ' +
-                     'for user {} and course {}'.format(ce.user.username,
-                                                        str(ce.course_id)))
-            )
-            log_error(
-                error_data=error_data,
-                error_type=PipelineError.COURSE_DATA,
-                user=ce.user,
-                course_id=str(course_id)
-            )
+        sm = student_modules_for_course_enrollment(
+            site=site,
+            course_enrollment=ce).order_by('-modified')
```

Comment:
I need to decide if it makes sense to filter out any SM records created after 'date_for'. At this point in the evolution of this code, SM is there only to check if there's been learner activity since the last LCGM record collected